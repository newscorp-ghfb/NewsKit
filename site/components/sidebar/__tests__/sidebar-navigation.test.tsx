import {
  renderToFragmentWithTheme,
  renderWithTheme,
} from '../../../utils/test-utils';
import {SidebarNav} from '..';

const useRouter = jest.spyOn(require('next/router'), 'useRouter');

const scrollIntoViewMock = jest.fn();
Element.prototype.scrollIntoView = scrollIntoViewMock;

jest.mock('next/link', () => ({children}: any) => children);
jest.mock('../../../routes', () => ({
  routes: [
    {
      title: 'Group 1',
      id: '/group1',
      subNav: [
        {
          title: 'Group 1 Page 1',
          page: true,
          id: '/group1/page1',
        },
      ],
    },
    {
      title: 'Group 2',
      id: '/group2',
      subNav: [
        {
          title: 'Group 2 Page 1',
          id: '/group2/page1',
          subNav: [
            {
              title: 'Group 2 Page 1 a',
              page: true,
              id: '/group2/page1/a',
            },
            {
              title: 'Group 2 Page 1 b',
              page: true,
              id: '/group2/page1/b',
            },
          ],
        },
        {
          title: 'Group 2 Page 2',
          id: '/group2/page2',
          subNav: [
            {
              title: 'Group 2 Page 2 a',
              page: true,
              id: '/group2/page2/a',
            },
            {
              title: 'Group 2 Page 2 b',
              page: true,
              id: '/group2/page2/b',
            },
          ],
        },
      ],
    },
  ],
}));

describe('Sidebar navigation', () => {
  test('should match snapshot with no active link', () => {
    useRouter.mockImplementationOnce(() => ({pathname: '/'}));

    expect(renderToFragmentWithTheme(SidebarNav)).toMatchSnapshot();
  });

  test('should render only routes under current section', () => {
    useRouter.mockImplementationOnce(() => ({pathname: '/group1'}));

    expect(renderToFragmentWithTheme(SidebarNav)).toMatchSnapshot();
  });

  test('should match snapshot with active link', () => {
    useRouter.mockImplementationOnce(() => ({pathname: '/group1/page1'}));
    expect(renderToFragmentWithTheme(SidebarNav)).toMatchSnapshot();
  });

  test('should not invoke scrollIntoView when link is inactive', () => {
    jest.clearAllMocks();
    useRouter.mockImplementationOnce(() => ({pathname: '/'}));
    renderWithTheme(SidebarNav);
    expect(scrollIntoViewMock).toHaveBeenCalledTimes(0);
  });
});
