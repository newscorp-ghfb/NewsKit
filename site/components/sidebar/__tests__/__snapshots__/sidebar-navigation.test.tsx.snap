// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar navigation should match snapshot with active link 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: "DM Sans",sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  letter-spacing: 0;
}

.emotion-1 {
  list-style-type: none;
  margin: 0px;
  padding: 0;
}

.emotion-1:last-child {
  margin-bottom: 24px;
}

.emotion-2 {
  margin-bottom: 24px;
}

.emotion-4 {
  display: inline-block;
  transition-property: color;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(0, 0, .5, 1);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4:hover:not(:disabled) {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-6 {
  font-family: "Poppins",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 600;
  letter-spacing: 0;
  padding: 0.5px 0px;
  background-color: transparent;
  color: #0A0A0A;
  box-sizing: border-box;
  cursor: 'pointer';
  min-width: 230px;
  padding: 12px;
  margin: 4px 0;
  text-overflow: ellipsis;
}

.emotion-6::before {
  content: '';
  margin-bottom: -0.403em;
  display: block;
}

.emotion-6::after {
  content: '';
  margin-top: -0.4em;
  display: block;
}

<nav
    aria-label="Sidebar"
    class="emotion-0"
    role="navigation"
  >
    <ul
      class="emotion-1"
      role="list"
    >
      <li
        class="emotion-2"
      >
        <div
          class="emotion-2"
        >
          <div>
            <a
              class="emotion-4"
              href="/group1/page1"
            >
              <span
                class="emotion-5"
              >
                <div
                  class="emotion-6"
                  data-testid="/group1/page1"
                >
                  Group 1 Page 1
                </div>
              </span>
            </a>
          </div>
        </div>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Sidebar navigation should match snapshot with no active link 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: "DM Sans",sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  letter-spacing: 0;
}

.emotion-1 {
  list-style-type: none;
  margin: 0px;
  padding: 0;
}

.emotion-1:last-child {
  margin-bottom: 24px;
}

<nav
    aria-label="Sidebar"
    class="emotion-0"
    role="navigation"
  >
    <ul
      class="emotion-1"
      role="list"
    />
  </nav>
</DocumentFragment>
`;

exports[`Sidebar navigation should render only routes under current section 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: "DM Sans",sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  letter-spacing: 0;
}

.emotion-1 {
  list-style-type: none;
  margin: 0px;
  padding: 0;
}

.emotion-1:last-child {
  margin-bottom: 24px;
}

.emotion-2 {
  margin-bottom: 24px;
}

.emotion-4 {
  display: inline-block;
  transition-property: color;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(0, 0, .5, 1);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4:hover:not(:disabled) {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-6 {
  font-family: "Poppins",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 600;
  letter-spacing: 0;
  padding: 0.5px 0px;
  background-color: transparent;
  color: #535353;
  box-sizing: border-box;
  cursor: 'pointer';
  min-width: 230px;
  padding: 12px;
  margin: 4px 0;
  text-overflow: ellipsis;
}

.emotion-6::before {
  content: '';
  margin-bottom: -0.403em;
  display: block;
}

.emotion-6::after {
  content: '';
  margin-top: -0.4em;
  display: block;
}

.emotion-6:hover:not(:disabled) {
  background-color: #E0E7FF;
  border-radius: 8px;
  color: #0A0A0A;
}

.emotion-6:active:not(:disabled) {
  background-color: #AEBFFF;
  border-radius: 8px;
  color: #0A0A0A;
}

<nav
    aria-label="Sidebar"
    class="emotion-0"
    role="navigation"
  >
    <ul
      class="emotion-1"
      role="list"
    >
      <li
        class="emotion-2"
      >
        <div
          class="emotion-2"
        >
          <div>
            <a
              class="emotion-4"
              href="/group1/page1"
            >
              <span
                class="emotion-5"
              >
                <div
                  class="emotion-6"
                  data-testid="/group1/page1"
                >
                  Group 1 Page 1
                </div>
              </span>
            </a>
          </div>
        </div>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
